src/SelectField.tsx

```diff
@@ -91,57 +93,70 @@ function isSelectOptionObject(
 }
 
 function RenderSelect(props: SelectInputProps) {
-  const selectDefault = props.fieldType === Array ? [] : props.placeholder;
-
   const [expanded, setExpanded] = useState<boolean>(false);
-  const [selected, setSelected] = useState<string | string[]>(selectDefault);
-
-  const handleSelect = (
-    event: React.MouseEvent | React.ChangeEvent,
-    selection: string | SelectOptionObject
-  ) => {
-    const items = parseInput(selection, props.fieldType);
-    props.onChange(items);
-    setSelected(items);
-    setExpanded(false);
-  };
-
-  const parseInput = (
-    selection: string | SelectOptionObject,
-    fieldType: typeof Array | any
-  ): string | string[] => {
-    const parsedSelection = isSelectOptionObject(selection)
-      ? selection.toString()
-      : selection;
-
-    if (fieldType !== Array) {
-      return parsedSelection !== '' ? parsedSelection : '';
-    }
-
-    if (Array.isArray(selected)) {
-      if (selected.includes(parsedSelection)) {
-        return selected.filter((s) => s !== parsedSelection);
+  const [selected, setSelected] = useState<string | string[]>([]);
+
+  const parseInput = useCallback(
```

Comment:
So happy with this, this is something that I was meaning to come back to, but haven't had the chance to do.